File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.cpp
#include <iostream>
#include <set>
#include <map>
#include <set>
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        map<int,int> hashmap;
        set<int> lookup;
        for (int i = 0; i<arr.size();i++){
            // find either iterator with key/val or iter the end
            if(hashmap.find(arr[i]) == hashmap.end()){
                hashmap[arr[i]] = 1;
            }
            else{
                hashmap[arr[i]]++;
            }
        }
        for (auto it = hashmap.begin(); it != hashmap.end();it++){
            lookup.insert(it->second);
        }
        return lookup.size() == hashmap.size();
    }
};

File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.ts
function uniqueOccurrences(arr: number[]): boolean {
    let lookup = new Map()
    for(let i = 0;i<arr.length;i++){
        if (lookup.has(arr[i])){
            lookup.set(arr[i],lookup.get(arr[i])+1)
        }
        else{
            lookup.set(arr[i],1)
        }
    }
    ///
    let seen = new Set();
    for (const val of lookup.values()){
        if (seen.has(val)){
            return false;
        }
        else {
            seen.add(val)
        }
    }
    return true;
};